base/src/plugin-manager/plugin-manager.ts

```diff
@@ -40,6 +40,10 @@ export class PluginManager {
     }
   }
 
+  static clearPlugins(): void {
```

Comment:
change this to onLogout, and make the class implement interface MVDHosting.LogoutActionInterface such that you can register ZoweZLUX.pluginManager here https://github.com/zowe/zlux-app-manager/blob/50e49e9b2313d8a79aa3315a4c9ff560cfa85d25/virtual-desktop/src/app/authentication-manager/authentication-manager.service.ts#L75 for inclusion in the logout process.

Reply:
tried doing this, it worked for the plugin manager in the zlux-app-manager fine, but for some reason it didn't work for this class it is not registrering the onLogout function properly and throwing an error saying its undefined